core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -1751,13 +1777,32 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
       new AlterConfigOp(new ConfigEntry(LogConfig.CleanupPolicyProp, LogConfig.Compact + "," + LogConfig.Delete), AlterConfigOp.OpType.SUBTRACT)
     ).asJavaCollection
 
-    alterResult = client.incrementalAlterConfigs(Map(
+    alterConfigs = Map(
       topic1Resource -> topic1AlterConfigs,
       topic2Resource -> topic2AlterConfigs
-    ).asJava)
+    )
+    alterResult = client.incrementalAlterConfigs(alterConfigs.asJava)
     assertEquals(Set(topic1Resource, topic2Resource).asJava, alterResult.values.keySet)
     alterResult.all.get
 
+    if (isKRaftTest()) {
```

Comment:
I think this can be a util function:
```
TestUtils.waitForMetadataPropagation(brokers, condition: KRaftMetadataCache => Boolean): Unit = {
  assertTrue(isKRaftTest())
  TestUtils.waitUntilTrue(() => {
    brokers.forall { broker => condition(broker.metadataCache.asInstanceOf[KRaftMetadataCache])
  }, "Metadata is not propgated to all brokers")
}
```

and we can call it as:
```
TestUtils.waitForMetadataPropagation(brokers, metadataCache => {
  // validation logic.
})
```

Reply:
I agree having some utilities would be nice. Rather than having something really specific, maybe we just need a utility which waits until the brokers have caught up to the controller end offset?

Reply:
Yeah, nice suggestion, waiting for the metadata offset is simpler and more generic. I will submit a separate PR to make this a util function since there are already many other places using this similar specific assertion.